policy/Policy.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Serilog;
+using Grpc.Core;
+
+namespace AKSMiddleware;
+
+public class LoggingPolicy : HttpPipelinePolicy
+{
+    private readonly Serilog.ILogger _logger;
+
+    public LoggingPolicy(Serilog.ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+    {
+        DateTime startTime = DateTime.UtcNow;
+
+        try
+        {
+            ProcessNext(message, pipeline);
+        }
+        catch (Exception ex)
+        {
+            _logger.Error(ex, "Exception occurred during HTTP processing.");
+            throw;
+        }
+        finally
+        {
+            Logging.LogRequest(new LogRequestParams(
+                _logger,
+                startTime,
+                message.Request,
+                message.Response,
+                null
+            ));
+        }
+    }
+
+    public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+    {
+        DateTime startTime = DateTime.UtcNow;
+
+        try
+        {
+            await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            _logger.Error(ex, "Exception occurred during HTTP processing.");
+            throw;
+        }
+        finally
+        {
+            // Log request details
+            Logging.LogRequest(new LogRequestParams(
+                _logger,
+                startTime,
+                message.Request,
+                message.Response,
+                null
+            ));
+        }
+    }
+}
+
+
+public static class ArmPolicy
+{
+    public static ArmClientOptions GetDefaultArmClientOptions(Serilog.ILogger logger)
+    {
+        var clientOptions = new ArmClientOptions
+        {
+            Retry =
+            {
+                MaxRetries = 5
+            }
+        };
+
+        clientOptions.AddPolicy(new LoggingPolicy(logger), HttpPipelinePosition.PerCall);
+
+        return clientOptions;
+    }
+
+    public static StatusCode ConvertHTTPStatusToGRPCError(int httpStatusCode)
```

Comment:
please ensure to leave a comment here with the link that explains the reasoning for the conversion (same as the go middleware)